### ISSUE TITLE
Gem takes too much space

### ISSUE THREAD

**Issue by DannyBen** [envkey/envkey-ruby#3]
This gem takes way too much space, due to all the files in the [ext folder][1].
I think this is unsustainable, and a different approach should be considered.

[1]: https://github.com/envkey/envkey-ruby/tree/master/ext

**Comment by danenania** [comment]
I agree it would be better to use less space. While I don't see how to get away from bundling binaries for each platform (seamless compatibility > file size imo), there are [some techniques](https://blog.filippo.io/shrink-your-go-binaries-with-this-one-weird-trick/) for reducing go binary sizes that seem worth exploring.

I'll give some of those a shot for compiling [envkey-fetch](https://github.com/envkey/envkey-fetch) and see if a significant reduction is possible.

**Comment by guingois** [comment]
What about not providing the binary at all, and only provide the Ruby wrapper that integrates it with a Ruby app ? This is fairly common pattern for gems that wrap binaries.

If you really think that your users should not have to deal with the binary at any point, then you could automate the installation by adding some code that download the right binary into the right folder.

I would argue that this feature isn't even necessary: the standalone installation of envkey-fetch is trivially easy and I would much prefer managing it myself once and for all tools, rather than doing it once because I need it outside my Ruby app, and having the gem install a duplicate (or in the current version, many useless duplicates) on its own. You could give your users a option to configure the path of the binary, and voilà, less work for the gem, more power to the users.

**Comment by danenania** [comment]
Thanks @erwanthomas -- the feedback is appreciated! I think a post-install hook that pulls down the appropriate binary is probably the best option.
